### PR TITLE
Revert "Cover: Fix matrix alignment issue."

### PR DESCRIPTION
**Pull request by ockham**
Reverts WordPress/gutenberg#28361

Reason: Visually broken on the frontend :crying_cat_face: 

See https://github.com/WordPress/gutenberg/pull/28361#issuecomment-763888663 for screenshots.

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Thanks Bernie.

I have an idea for an alternate fix I'm going to try in a moment.

**Comment by jasmussen** [comment]
I created https://github.com/WordPress/gutenberg/pull/28365. At this point I'm not sure whether we should land it as a point release, or get land this revert and look for the root cause that made the image shift around when using the matrix alignment.

**Comment by ockham** [comment]
> I created #28365. At this point I'm not sure whether we should land it as a point release, or get land this revert and look for the root cause that made the image shift around when using the matrix alignment.

Thanks! I'm thinking to land the the revert and ship the point release with it, which basically gets us back to the state of Cover block styling prior to #28114, which is a good enough baseline -- mostly to derisk it, since at this point, I'm a bit wary that any other CSS changes might have other unwanted side effects.

We can then merge #28365 to `master` (but not to the 9.8 branch) and see if it needs any further tweaking there.

**Comment by ockham** [comment]
Hmm, really still isn't exactly pretty on the frontend:

image

Thinking of holding off on 9.8.1 a bit more to either include #28365 after all, and/or, ideally, find the root cause of the visual regression. @jasmussen would you have any spare cycles to look into the latter? :blush:

**Comment by jasmussen** [comment]
That shift you show in the screenshot is what https://github.com/WordPress/gutenberg/pull/28365 is meant to fix.

I'm juggling some toddlers at the moment but can look later, and I believe what remains to be done is

A) figure out when and how that matrix issue regressed in the first place
B) either push a fix based on what we learn there, or embrace the fix already made in https://github.com/WordPress/gutenberg/pull/28365
C) test that fix super thoroughly

**Comment by jasmussen** [comment]
In my highly un-scientific shotgun debugging session, as an alternative to bisecting, I went to the list of [commits](https://github.com/WordPress/gutenberg/commits/master), went pages and pages back, checked out and old version and `npm ci && npm run dev` 'ed it. I used the Twenty Twenty theme, and this is what master looks like today:

latest master

First time I went far enough back, to December 15th, that the block was still using background images. In that version, it worked fine.

dec15

Then I checked out this changeset: https://github.com/WordPress/gutenberg/pull/25171 and the bug was back again:



So I checked out the commit right before it, and the bug was gone.

right before

jan13

So yes, it appears that we forgot to test the matrix alignment, when the Cover was refactored to use an image with srcsets instead. The take-away is that there wasn't ever a time after the srcset refactor, where the matrix alignment worked perfectly, so there isn't any mystical CSS regression to find. Which suggests we can keep exploring fixes like the one in #28365. And I still believe that fix is solid, simple, basic, and should work. Because of that, I'd recommend the next step be to test that thoroughly, then decide how and when to ship it.

**Comment by ajlende** [comment]
I have a branch that seems to be working fine before the revert: https://github.com/ajlende/gutenberg/tree/fix/cover-alignment

**Comment by jasmussen** [comment]
> I have a branch that seems to be working fine before the revert:

Did you find out what was unique to that branch, making it work?

I'm about to dive in and test your other branch. Thanks for all your work.

**Comment by ajlende** [comment]
> Did you find out what was unique to that branch, making it work?

Yep, and it's all in the other branch already.

**Pull request merged**
